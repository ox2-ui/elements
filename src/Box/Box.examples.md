### Box

    <Box customStyle={{ padding: '30px' }}>
      <div>Child</div>
    </Box>
