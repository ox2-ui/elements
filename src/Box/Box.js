import React, { PropTypes } from 'react';

/**
 * Box Component
 */
const Box = ({ className, style, children, customStyle }) => (
  <div
    className={className}
    style={{ ...style, ...customStyle }}
  >
    {children}
  </div>
);


Box.propTypes = {
  /**
   * Box content.
   */
  children: PropTypes.node.isRequired,
  /**
   * The css class name of the root element.
   */
  className: PropTypes.string,
  /**
   * Box specific style properties.
   */
  customStyle: PropTypes.object.isRequired,
  /**
   * Override the inline-styles of the text element.
   */
  style: PropTypes.object,
};

export default Box;
