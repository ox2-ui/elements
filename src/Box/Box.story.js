import React from 'react';
import { storiesOf } from '@kadira/storybook';
import Box from './Box';

storiesOf('Box', module)
  .add('default', () => (
    <Box customStyle={{ padding: '30px' }}>
      <div>Child</div>
    </Box>
  ));
