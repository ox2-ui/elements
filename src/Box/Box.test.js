/* eslint-disable no-unused-vars */
import React from 'react';
import { shallow, render, mount } from 'enzyme';
import Box from './Box';

const handler = () => true;

it('renders correctly', () => {
  const wrapper = mount(
    <Box customStyle={{ padding: '30px' }}>
      <div>Child</div>
    </Box>
  );

  expect(wrapper).toMatchSnapshot();
});
