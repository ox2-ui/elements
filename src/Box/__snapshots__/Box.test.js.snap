// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<Box
  customStyle={
    Object {
      "padding": "30px",
    }
  }
>
  <div
    style={
      Object {
        "padding": "30px",
      }
    }
  >
    <div>
      Child
    </div>
  </div>
</Box>
`;
